background/background.js

```diff
@@ -185,20 +185,40 @@ const ContextMenu = {
   }
 };
 
-LivemarkUpdater.init().then(() => ContextMenu.init());
+// This value needs to be synchronously accessible in the
+// chrome.webRequest.onHeadersReceived listener below.
+let feedPreviewEnabled = true;
+async function feedPreview() {
```

Comment:
Thinking of this, it would be nice to have a FeedPreview namespace with two methods: init() and show(tabId, url), and an `enabled` property.

Then:

- feedPreview() -> FeedPreview.init()
- redirectToSubscribe() -> FeedPreview.show()
- feedPreviewEnabled -> FeedPreview.enabled

You can define FeedPreview the same way LivemarkUpdater and ContextMenu are defined.